lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -969,8 +971,20 @@ data ByronWalletPutPassphraseData = ByronWalletPutPassphraseData
     , newPassphrase :: !(ApiT (Passphrase "user"))
     } deriving (Eq, Generic, Show)
 
+data ApiSealedTxEncoding = HexEncoded | Base64Encoded
+      deriving (Eq, Generic, Show)
+      deriving anyclass NFData
+
+instance ToJSON ApiSealedTxEncoding where
+    toJSON HexEncoded = Bool True
+    toJSON Base64Encoded = Bool False
+instance FromJSON ApiSealedTxEncoding where
+    parseJSON = withBool "ApiSealedTxEncoding" $ \b -> case b of
+        True -> pure HexEncoded
+        False -> pure Base64Encoded
```

Comment:
```suggestion
    parseJSON = withBool "ApiSealedTxEncoding" $ 
        pure . bool Base64Encoded HexEncoded
```
?

Reply:
I'm going to call the `bool`ice on `bool`.

If you really want to get rid of the intermediate name `b`, I would prefer.
```
parseJSON = withBool "ApiSealedTxEncoding" . pure $ \case
    True -> HexEncoded
    False -> Base64Encoded
```